### ISSUE TITLE
Plot panel reorder topics by dragging

### ISSUE THREAD

**Issue by Combinacijus** [foxglove/studio#4603]
I would like to reorder topics in Plot panel but can't find how
![image](https://user-images.githubusercontent.com/16375702/194767337-5c97f3ef-469d-410f-bb0c-3c4af267823c.png)

**Comment by jtbandes** [comment]
This is a good request. I would add that we are more likely to implement it in the sidebar settings (https://github.com/foxglove/studio/pull/4321), since we are exploring whether to replace the editable legend with this sidebar. cc @2metres @foxymiles

**Comment by foxymiles** [comment]
Yeah I agree if we're going to do something like this it would make sense as part of the migration to the sidebar. Then other panels could benefit from the work.

**Comment by jtbandes** [comment]
We've also already implemented this kind of thing in the sidebar in the Indicator panel (although doesn't support drag-and-drop). It would be nice if this were an inherent feature provided by the sidebar settings API in a reusable way:

<img width="575" alt="image" src="https://user-images.githubusercontent.com/14237/195464245-3c48252c-e8bf-443f-bedf-5445417b0515.png">
